cpp/src/arrow/acero/task_util_test.cc

```diff
@@ -231,5 +231,96 @@ TEST(TaskScheduler, StressTwo) {
   }
 }
 
+TEST(TaskScheduler, AbortContOnTaskErrorSerial) {
+  constexpr int kNumTasks = 16;
+
+  auto scheduler = TaskScheduler::Make();
+  auto task = [&](std::size_t, int64_t task_id) {
+    if (task_id == kNumTasks / 2) {
+      return Status::Invalid("Task failed");
+    }
+    return Status::OK();
+  };
+
+  int task_group =
+      scheduler->RegisterTaskGroup(task, [](std::size_t) { return Status::OK(); });
+  scheduler->RegisterEnd();
+
+  ASSERT_OK(scheduler->StartScheduling(
+      0, [](TaskScheduler::TaskGroupContinuationImpl) { return Status::OK(); }, 1, true));
```

Comment:
/Can we annotate non-obvious parameters with their names?
```suggestion
  ASSERT_OK(scheduler->StartScheduling(
      /*xxx=*/0, [](TaskScheduler::TaskGroupContinuationImpl) { return Status::OK(); },
      /*xxx=*/1, /*xxx=*/true));
```

Reply:
Thank you. Done.